src/test/java/com/team01/carsharingapp/service/CarServiceTests.java

```diff
@@ -0,0 +1,246 @@
+import static org.mockito.Mockito.when;
+
+import com.team01.carsharingapp.dto.car.request.CreateCarRequestDto;
+import com.team01.carsharingapp.dto.car.response.CarDto;
+import com.team01.carsharingapp.exception.EntityNotFoundException;
+import com.team01.carsharingapp.mapper.CarMapper;
+import com.team01.carsharingapp.model.Car;
+import com.team01.carsharingapp.repository.CarRepository;
+import com.team01.carsharingapp.service.impl.CarServiceImpl;
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+public class CarServiceTests {
+    private static final int ONCE = 1;
+    private static final Long ID_ONE = 1L;
+    private static final Long INCORRECT_ID = 42L;
+    private static final String CAR_NOT_FOUND_MESSAGE = "Can't find car by id: ";
+    private static final String TEST_CAR_MODEL = "City";
+    private static final String TEST_CAR_BRAND = "Honda";
+    private static final Car.Type TEST_CAR_TYPE = Car.Type.valueOf("SEDAN");
+    private static final int TEST_CAR_AMOUNT = 3;
+    private static final BigDecimal TEST_CAR_FEE = BigDecimal.valueOf(5);
+    private static final String TEST_DTO_MODEL = "RAV4";
+    private static final String TEST_DTO_BRAND = "TOYOTA";
+    private static final String TEST_DTO_TYPE = "SUV";
+    private static final int TEST_DTO_AMOUNT = 1;
+    private static final BigDecimal TEST_DTO_FEE = BigDecimal.valueOf(3);
+
+    @Mock
+    private CarRepository carRepository;
+    @Mock
+    private CarMapper carMapper;
+
+    @InjectMocks
+    private CarServiceImpl carService;
+
+    @Test
+    @DisplayName("Save car by correct data")
+    public void save_AllCorrectData_CorrectDtoReturned() {
+        CreateCarRequestDto requestDto = createValidCreateCarRequestDto();
+        Car expectedCarWithoutId = getCarFromCreateCarRequestDto(requestDto);
+        Car expectedCarWithId = getCarFromCreateCarRequestDto(requestDto);
+        expectedCarWithId.setId(ID_ONE);
+        CarDto expected = getCarDtoFromCar(expectedCarWithId);
+
+        when(carMapper.toEntity(requestDto)).thenReturn(expectedCarWithoutId);
+        when(carRepository.save(expectedCarWithoutId)).thenReturn(expectedCarWithId);
+        when(carMapper.toDto(expectedCarWithId)).thenReturn(expected);
+
+        CarDto actual = carService.save(requestDto);
+
+        Assertions.assertEquals(expected, actual);
+        verify(carRepository, times(ONCE)).save(expectedCarWithoutId);
+        verifyNoMoreInteractions(carRepository);
+        verify(carMapper, times(ONCE)).toEntity(requestDto);
+        verify(carMapper, times(ONCE)).toDto(expectedCarWithId);
```

Comment:
times(1) used by default and it's not necessary to use verify() if we checks method in WHEN